src/index.js

```diff
@@ -0,0 +1,284 @@
+import StatusApi from './statusApi.js';
+import {buildUrl} from './utils';
+const Ip = require('ip');
+const fs = require('fs');
+const path = require('path');
+
+// Duration (in ms) to wait before attempting to reconnect.
+const CONNECT_INTERVAL = 4000;
+const CHAIN_NAME = 'embark';
+const NETWORK_NAME = 'Embark';
+
+// private method symbols
+const _connectStatus = Symbol('connectStatus');
+const _connectToStatusNetwork = Symbol('connectToStatusNetwork');
+const _addAndConnectStatusNetwork = Symbol('addAndConnectStatusNetwork');
+const _getStatusNetworkId = Symbol('getStatusNetworkId');
+const _setStatusNetworkId = Symbol('setStatusNetworkId');
+const _registerEvents = Symbol('registerEvents');
+
+/**
+ * Plugin that connects an Embark dApp to the Status app, and allows the dApp
+ * to be run in the Status browser.
+ */
+class EmbarkStatusPlugin {
+  constructor(embark) {
+    this.networkId = 1337;
+    this.embark = embark;
+    this.events = this.embark.events;
+    this.pluginConfig = this.embark.pluginConfig;
+    this.deviceIp = this.pluginConfig.deviceIp;
+    this.logger = this.embark.logger;
+    // we want a timeout to occur before issuing the next request, so reduce the timeout specified
+    this.statusApi = new StatusApi({ip: this.deviceIp, logger: this.logger, timeout: CONNECT_INTERVAL - 1000});
+    this.webServerConfig = {};
+    this.blockchainConfig = {};
+    this.networkIdDirectory = path.join(process.env.DAPP_PATH, '.embark', 'embark-status');
+    this.networkIdPath = path.join(this.networkIdDirectory, 'networkId');
+
+    // gets hydrated webserver config from embark
+    this.events.on('config:load:webserver', webServerConfig => {
+      this.webServerConfig = webServerConfig;
+    });
+
+    // gets hydrated blockchain config from embark
+    this.events.on('config:load:blockchain', blockchainConfig => {
+      this.blockchainConfig = blockchainConfig;
+    });
+
+    // adds cors to blockchain and storage clients
+    this.events.request('config:cors:add', `http://${this.deviceIp}`);
```

Comment:
Noob speaking here:

Can we just assume that this is `http` and not `https` in the future?

Reply:
I think this is dependent upon the webserver connection on the app, and as far as I know this is `http` only for now. I guess man-in-the-middle attacks would be difficult given that the comms are localised to a local network (the connection is from a host machine running embark to a phone on the same LAN, and the HTTP comms go from machine to phone). I would say this is OK for now.